lib/ram_pwrdn/ram_pwrdn.c

```diff
@@ -121,7 +157,7 @@ static uintptr_t ram_end_addr(void)
 {
 	const struct ram_bank *last_bank = &banks[RAM_BANK_COUNT - 1];
 
-	return last_bank->start + ram_bank_size(last_bank);
+	return MIN((last_bank->start + ram_bank_size(last_bank)), RAM_APP_END_ADDR);
```

Comment:
I think we may safely assuemt that `RAM_APP_END_ADDR <= last_bank->start + ram_bank_size(last_bank)` so I would just use RAM_APP_END_ADDR.

Reply:
Will return `last_bank->start + ram_bank_size(last_bank)` if partition manager is not enabled or else `PM_SRAM_PRIMARY_END_ADDRESS`. `RAM_APP_END_ADDR` will no longer be needed if DTS is not used.